### PR TITLE
Move from MetadataUtils to Device

### PR DESCRIPTION
**Pull request by nelsitoPuglisi**
## Goal

Avoid using legacy dataclass `DeviceInfo` and static `MetadataUtils`

Fixes typo on *EGL* `(Embedded-System Graphics Library)`

## Testing

Unit tests

### CODE REVIEW

**Comment by nelsitoPuglisi** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/546?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#546** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/546?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#517** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/517?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#516** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/516?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#515** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/515?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#514** <a href="https://app.graphite.dev/github/pr/embrace-io/embrace-android-sdk/514?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @nelsitoPuglisi and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/payload/EnvelopeResource.kt [hunk 1]
```diff
@@ -176,9 +176,18 @@ internal data class EnvelopeResource(
...
     @Json(name = "num_cores")
-    val numCores: Int? = null
-
-) {
+    val numCores: Int? = null,
+
+    /* (Android) The name of the primary CPU of the device, obtained with the system call 'ro.board.platform'.
+    Previous name: d.pt */
+    @Json(name = "cpu_name")
+    val cpuName: kotlin.Int? = null,
```
Comment (fractalwrench) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/payload/EnvelopeResource.kt:184]:
```suggestion
    val cpuName: Int? = null,
```

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/payload/EnvelopeResource.kt [hunk 2]
```diff
@@ -176,9 +176,18 @@ internal data class EnvelopeResource(
...
+
+    /* (Android) The EGL (Embedded-System Graphics Library) information obtained with the system call
+    'ro.hardware.egl'. Previous name: d.gp */
+    @Json(name = "egl_info")
+    val eglInfo: kotlin.Int? = null
```
Comment (fractalwrench) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/payload/EnvelopeResource.kt:189]:
```suggestion
    val eglInfo: Int? = null
```

File: embrace-android-sdk/src/test/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImplTest.kt [hunk 3]
```diff
@@ -1,19 +1,58 @@
...
+
+        @BeforeClass
+        @JvmStatic
+        fun beforeClass() {
+            mockkStatic(Environment::class)
+
+            initContext()
+
+            every { Environment.getDataDirectory() }.returns(File("ANDROID_DATA"))
+            //every { MetadataUtils.getInternalStorageFreeCapacity(any()) }.returns(123L)
```
Comment (fractalwrench) [inline embrace-android-sdk/src/test/java/io/embrace/android/embracesdk/capture/envelope/resource/EnvelopeResourceSourceImplTest.kt:33]: this commented out code can be deleted

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/Device.kt [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+import androidx.annotation.ChecksSdkIntAtLeast
+import io.embrace.android.embracesdk.capture.cpu.CpuInfoDelegate
+import io.embrace.android.embracesdk.logging.InternalStaticEmbraceLogger.Companion.logDebug
+import io.embrace.android.embracesdk.logging.InternalStaticEmbraceLogger.Companion.logDeveloper
+import io.embrace.android.embracesdk.prefs.PreferencesService
+import io.embrace.android.embracesdk.worker.BackgroundWorker
+import java.io.File
+import java.util.Locale
+
+public interface Device {
```
Comment (fractalwrench) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/Device.kt:17]: This and all its fields can be `internal` unless there's a good reason to expose any of this

File: embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/Device.kt [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+     */
+    public val numberOfCores: Int
+
+    /**
+     * Gets the free capacity of the internal storage of the device.
+     *
+     * @param statFs the {@link StatFs} service for the device
+     * @return the total free capacity of the internal storage of the device in bytes
+     */
+    public val internalStorageTotalCapacity: Lazy<Long>
```
Comment (fractalwrench) [inline embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/capture/envelope/resource/Device.kt:83]: nit: this should probably be type `Long` in the interface, and the implementer can choose to implement it as `Lazy`. Not a big deal but it'll make fakes marginally easier to write

**Pull request merged**